### PR TITLE
python3Packages.opentelemetry-instrumentation-botocore: init at 0.46b0

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

Add opentelemetry-instrumentation-botocore python module https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-botocore

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by onny** [comment]
Result of `nixpkgs-review pr 326504` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.opentelemetry-instrumentation-botocore</li>
    <li>python311Packages.opentelemetry-instrumentation-botocore.dist</li>
    <li>python311Packages.opentelemetry-propagator-aws-xray</li>
    <li>python311Packages.opentelemetry-propagator-aws-xray.dist</li>
    <li>python312Packages.opentelemetry-instrumentation-botocore</li>
    <li>python312Packages.opentelemetry-instrumentation-botocore.dist</li>
    <li>python312Packages.opentelemetry-propagator-aws-xray</li>
    <li>python312Packages.opentelemetry-propagator-aws-xray.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  build-system = [ hatchling ];
+
+  dependencies = [
+    opentelemetry-api
+    opentelemetry-instrumentation
+    opentelemetry-semantic-conventions
+    opentelemetry-propagator-aws-xray
+  ];
+
+  nativeCheckInputs = [ botocore ];
```
Comment (natsukium) [inline pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix:36]:
```suggestion
  nativeCheckInputs = [
    botocore
    pytestCheckHook
  ];
```
Could you add `pytestCheckHook` to set the checkPhase?

File: pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  moto,
+  opentelemetry-instrumentation-botocore,
+  opentelemetry-test-utils,
+  opentelemetry-propagator-aws-xray,
+}:
+
+buildPythonPackage rec {
+  inherit (opentelemetry-instrumentation) version src;
+  pname = "opentelemetry-instrumentation-botocore";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix:21]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/opentelemetry-propagator-aws-xray/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  disabled = pythonOlder "3.8";
+
+  sourceRoot = "${opentelemetry-instrumentation.src.name}/propagator/opentelemetry-propagator-aws-xray";
+
+  build-system = [ hatchling ];
+
+  dependencies = [ opentelemetry-api ];
+
+  checkInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/opentelemetry-propagator-aws-xray/default.nix:26]:
```suggestion
  nativeCheckInputs = [
```

File: pkgs/development/python-modules/opentelemetry-propagator-aws-xray/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    pytest-benchmark
```
Comment (natsukium) [inline pkgs/development/python-modules/opentelemetry-propagator-aws-xray/default.nix:27]:
```suggestion
```
Can we remove the benchmark plugin?

File: pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  passthru.optional-dependencies = {
+    instruments = [
+      botocore
+    ];
+    test = [
+      markupsafe
+      moto
+      opentelemetry-instrumentation-botocore
+      opentelemetry-test-utils
+    ];
```
Comment (natsukium) [inline pkgs/development/python-modules/opentelemetry-instrumentation-botocore/default.nix:47]:
```suggestion
```
Should these dependencies go to `nativeCheckInputs`?
https://github.com/open-telemetry/opentelemetry-python-contrib/blob/38146dd7b4bf8dd35822b218b7cc9531208fdc77/instrumentation/opentelemetry-instrumentation-botocore/pyproject.toml#L33-L36

**Pull request closed without merging**